lib/addons/upbound-universal-crossplane/index.ts

```diff
@@ -1,36 +1,96 @@
-import { CoreAddOn } from "../core-addon";
-import { supportsX86 } from "../../utils";
+import 'source-map-support/register';
+import { Construct } from 'constructs';
+import {ClusterInfo, Values} from "../../spi";
+import { merge } from "ts-deepmerge";
+import {createNamespace, supportsALL} from '../../utils';
+import { Policy, PolicyDocument} from 'aws-cdk-lib/aws-iam';
+import * as cdk from 'aws-cdk-lib';
+import {HelmAddOn, HelmAddOnProps, HelmAddOnUserProps} from "../helm-addon";
 
 /**
- * Interface for Upbound Universal Crossplane EKS add-on options
+ * User provided options for the Helm Chart
  */
-interface UpboundUniversalCrossplaneAddOnProps {
+export interface UpboundCrossplaneAddOnProps extends HelmAddOnUserProps {
     /**
-     * Version of the driver to deploy
+     * To Create Namespace using CDK
      */
-    version?: string;
+    createNamespace?: boolean;
+    /*
+     * EKS MasterRole. `mastersRole` of blueprints should be passed to this parameter.
+     */
+    eksMasterRole?: any;

```

Comment:
Why does it need this master role? I assume it needs role of the clusters that crossplane will manage, not the management cluster which runs crossplane. 

Another question - why is type `any` as opposed to let's say `IRole` or string for arn?

Reply:
I agree and im using `String` which can be used on the role name.

Reply:
Can it be cross-account for real use cases? E.g. in a multi-account architecture, how do you see that role passed? We could make it IRole and allow customers either to create a role or look up role (supporting cross account). 

If it is string, then no validation is applied upfront. They can pass anything, and until the attempt something later through crossplane they will see no failure. 

Another option is for the string to be role arn (I can settle for this). The most generic option is `role: IRole | string`.